rllib/algorithms/algorithm_config.py

```diff
@@ -2556,6 +2560,27 @@ def offline_data(
                 `MultiAgentEpisode` not supported, yet). Note,
                 `rllib.core.columns.Columns.OBS` will also try to decompress
                 `rllib.core.columns.Columns.NEXT_OBS`.
+            materialize_data: If the raw data should be materialized in memory. This
+                boosts performance extensively, but needs enough memory to avoid an OOM.
+                Make sure that your cluster has the resources available. For very large
+                data you might want to switch to streaming mode, i.e. `False`. The
+                default is `False`. If your algorithm does not need the module in its
+                connectors (`ConnectorV2` pipeline) or all connectors are stateless
+                you might want to consider setting `materialize_mapped_data` to `True`
+                instead and set `materialize_data` to `False`. If your data does not fit
+                into memory set both configurations to `False`.
+            materialize_mapped_data: If the data should be materialized after running it
+                through the learner connectors (i.e. after running the
+                `OfflinePreLearner`). This improves performance extensively, but should
+                only be applied, if the connectors do not depend on the `RLModule` or
+                keep a state (e.g. MARWIL computes the values with a module). In the
+                latter case the training batches become stale after some iterations and
+                learning degrades or diverges. Ensure that your cluster has enough
+                resources available to avoid an OOM. If set to `True` make sure that
+                `materialize_data` is set to `False` to avoid materializing two
+                datasets. The default is `True`. If your data does not fit into memory
+                set both configurations (i.e. `materialize_data` and
+                `materialize_mapped_data`) to `False`.
```

Comment:
```suggestion
            materialize_mapped_data: Whether the data should be materialized after
                running it through the Learner connector pipeline (i.e. after running the
                `OfflinePreLearner`). This improves performance, but should only be used
                in case the (learner) connector pipeline does not require an RLModule
                and the (learner) connector pipeline is stateless. For example, MARWIL's
                Learner connector pipeline requires the RLModule for value function
                predictions and training batches would become stale after some iterations
                causing learning degradation or divergence. Also ensure that your cluster
                has enough memory available to avoid an OOM. If set to `True` (True), make
                sure that `materialize_data` is set to `False` to avoid materialization of two
                datasets.
                If your data does not fit into memory and your Learner connector pipeline
                requires an RLModule or is stateful, set both
                `materialize_data` and `materialize_mapped_data` to `False`.
```